### DISCUSSION TITLE
Error 403 Forbidden

### DISCUSSION THREAD

**Discussion by Nat3J0hns0n** [snipe/snipe-it#13043] [Q&A] [answered]
Any help would be appreciated, I have posted twice on discord, and have gotten 0 responses, so I am open for ideas at this point.

It is a Rocky 8 vm, that is hardened.

I am trying to go to the site via IP, and all it shows is Forbidden You don't have permission to access this resource.  The entire snipe-it folder is set to 755 for permissions.  And the user owner is snipeit while apart of the apache group.  I am unsure on what to do next, and I have followed all steps and had issues along the way where composer would not work, but I got that resolved, and it did all that it needed to do that I am aware of.
This is what shows up for the composer.  

![Screen Shot 2023-05-17 at 11 03 53 AM](https://github.com/snipe/snipe-it/assets/96807879/2930ff1e-1c26-4957-91f4-3f59819c5ea0)

Here is my snipe-it.conf file as well: (IP blocked out, but it is just the IP, not http:// or anything else.
 
![Screen Shot 2023-05-17 at 11 06 57 AM](https://github.com/snipe/snipe-it/assets/96807879/a19e647e-94a9-4006-9322-bf484f77061f)

**Comment by marcusmoore** [comment]
Hi there @Nat3J0hns0n 👋🏾 

Can you change `DocumentRoot` to `/var/www/snipe-it/public` and see if that works?

More details can be found in the [Linux server configuration docs](https://snipe-it.readme.io/docs/linuxosx).

**Reply by Nat3J0hns0n** [reply]
Hi @marcusmoore thank you for the suggestion, I wet ahead and tried that, and I have tried this in the past, and I got what you see in the screenshot.  The .htaccess file has snipeit as owner, and apache as group.  And permissions are set to 644.  So that it is readable.
![Screen Shot 2023-05-17 at 11 26 17 AM](https://github.com/snipe/snipe-it/assets/96807879/6bd175bf-1021-4a08-af4b-32adb8873050)

**Reply by marcusmoore** [reply]
Any chance you are using Apache 2.4? The docs say you may need to use `Require all granted` instead of `Allow From All` if so 🤔

**Reply by Nat3J0hns0n** [reply]
I will try that out real quick

**Reply by Nat3J0hns0n** [reply]
Still the same issue from that screenshot

**Reply by marcusmoore** [reply]
Which user is running apache? And I have to ask to be sure: you're restarting apache after making changes to the configuration right? 😄

**Reply by Nat3J0hns0n** [reply]
The httpd in /etc is owned by root.  And yes I am restarting httpd every time that I make a update to any .conf

**Reply by Nat3J0hns0n** [reply]
I have also been setting up a snipe-it instance on a non hardened vm, and I think it is running better, I had a SELinux troubleshooter pop up (you can see in the image), and it was telling me to run `sudo setsebool -P httpd_can_network_connect 1`. and `sudo setsebool -P httpd_can_network_connect_db 1`.  I ran both of those, then restarted httpd, and now I just have a white page when going to the ip that the webpage is hosted on.  Not sure what else to do from here, since it does not give me a SELinux troubleshooter pop up anymore

**Reply by snipe** [reply]
We do have some docs on how to set permissions correctly with SELinux:

https://snipe-it.readme.io/docs/common-issues#swift_transportexception-connection-could-not-be-established-with-host-permission-denied-13
https://snipe-it.readme.io/docs/linuxosx#notes-on-selinux

**Reply by marcusmoore** [reply]
With just a white page showing it is possible that the directories Laravel needs to write to are unwritable. The docs linked above should cover that aspect at least.

**Reply by snipe** [reply]
Yeah, Laravel can behave really oddly if it can't write its own errors to the log file. (You'll also want to check your php log and your apache log for possible additional context)

**Reply by Nat3J0hns0n** [reply]
@snipe where are those logs at?

**Reply by Nat3J0hns0n** [reply]
I have also run all those commands and none of them changed the error exception I am getting with debug enabled
![Screen Shot 2023-05-17 at 1 55 45 PM](https://github.com/snipe/snipe-it/assets/96807879/6ea66891-3902-4c00-b684-8384b992f6e2)

**Reply by snipe** [reply]
Again, these are all permissions issues. It can't write to the storage directory (or subfolders).

**Reply by Nat3J0hns0n** [reply]
also in the chrome console I get `Failed to load resource: the server responded with a status of 500 (Internal Server Error)`

**Reply by Nat3J0hns0n** [reply, marked as answer]
I thought that the storage directory had the right permissions, but it turns out I did 775, compared to 757.  And that worked
